streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -788,7 +788,15 @@ void runOnce() {
              */
             do {
                 log.debug("Processing tasks with {} iterations.", numIterations);
-                final int processed = taskManager.process(numIterations, time);
+                final int processed;
+                try {
+                    processed = taskManager.process(numIterations, time);
+                } catch (final Exception e) {
+                    log.error("encountered an error when processing tasks." +
+                        " Will commit all previously successfully processed tasks", e);
+                    taskManager.commitSuccessfullyProcessedTasks();
```

Comment:
nit: why not just keep this logic within `taskManager#process` and make `#commitSuccessfullyProcessedTasks` private?